htlcswitch/interceptable_switch.go

```diff
@@ -139,7 +139,16 @@ func (f *interceptedForward) Resume() error {
 
 // Fail forward a failed packet to the switch.
 func (f *interceptedForward) Fail() error {
-	reason, err := f.packet.obfuscator.EncryptFirstHop(lnwire.NewTemporaryChannelFailure(nil))
+	update, err := f.htlcSwitch.cfg.FetchLastChannelUpdate(
+		f.packet.incomingChanID,
```

Comment:
Yeah I think this is the proper link to use here, as depending on the use case, an outgoing edge might not even actually exist (hence the need for interception). In theory we could extend the API to allow the caller to apss in their own failure, but that's a larger change outside the scope of this simple bug fix.

Reply:
If the sender is an `lnd` node, then sending back a fresh channel update will trigger the "second chance" logic which'll result in retry attempt. The prior behavior of not sending back a channel update at all would cause some other implementations to halt w/ a terminal failure as they were expecting an update to be included.

Reply:
Yeah, this is the best strategy imo. roasbeef is spot on with the reasoning.